iree/compiler/Dialect/Flow/Transforms/test/unroll_reductions.mlir

```diff
@@ -0,0 +1,24 @@
+// RUN: iree-opt -split-input-file -iree-flow-unroll-reductions -cse %s | IreeFileCheck %s
+
+// CHECK-LABEL: func @unrolled_reduction
+func @unrolled_reduction(%arg0: tensor<4x2x8xf32>) -> tensor<4xf32> {
+  // CHECK-DAG: [[INITIAL:%.+]] = constant dense<0.000000e+00> : tensor<f32>
+  %cst = constant dense<0.000000e+00> : tensor<f32>
+  // CHECK-NEXT: [[TEMP:%.+]] = "xla_hlo.reduce"(%arg0, [[INITIAL]]) ( {
+  // CHECK-NEXT: ^bb0(%arg1: tensor<f32>, %arg2: tensor<f32>):	// no predecessors
+  // CHECK-NEXT:   %2 = xla_hlo.add %arg1, %arg2 : tensor<f32>
+  // CHECK-NEXT:   "xla_hlo.return"(%2) : (tensor<f32>) -> ()
```

Comment:
Is it necessary for the reduction function to be defined in each of the unrolled `xla_hlo.reduce` ops? Do they get de-duped later?

Reply:
Yep - the op needs the region. You have to outline to get any deduping, which is what we're moving away from (and relying on backends to fold/dedupe if needed).